src/clj/game/cards/identities.clj

```diff
@@ -890,6 +890,38 @@
                                 (trigger-event state side :ice-subtype-changed ice)))}]
     :events {:run-ends nil}}
 
+   "Saraswati Mnemonics: Endless Exploration"
+   (letfn [(install-card [chosen]
+                        {:prompt "Select a remote server"
+                         :choices (req (conj (vec (get-remote-names state)) "New remote"))
+                         :async true
+                         :effect (req (corp-install state side (assoc chosen :advance-counter 1) target)
+                                      (effect-completed state side eid)
```

Comment:
You should probably move the `corp-install` to the end of the `install-card` function, then you can drop the `effect-completed`. Unless `register-turn-flag!` gets unset during the install.

Something like:
```
 (req (let [tgt...]
          ...
         )
        (corp-install state side eid (assoc ...) target))
```

Note passing the `eid` into `corp-install`, that's what lets you drop the `effect-completed`.

Reply:
John has the right idea. Currently, you will trigger `corp-install` (which will show a "Choose a server dialog"), and then because you are not `wait-for`-ing that call, you will mark your ability as resolved immediately. It's not a huge deal because this is an `:ability` and not an event handler, but let's do it proper like! :)

Reply:
Refactored as described, but the `eid` argument got the compiler to spit errors. I tried just moving the position of `corp-install` without it and everything *seems* to work as intended. Is this alright or is there potential for a hidden issue?